ast/Trees.h

```diff
@@ -56,6 +56,23 @@ struct ParsedFile {
     core::FileRef file;
 };
 
+/**
+ * Stores the output of an AST pass. May be empty if pass was canceled or encountered an error.
+ * TODO: Modify to store reason if we ever have multiple reasons for a pass to stop. Currently, it's only empty if the
+ * pass is canceled in LSP mode.
+ */
+class MaybeASTPassResult final {
+private:
+    std::optional<std::vector<ParsedFile>> trees;
+
+public:
+    MaybeASTPassResult();
+    MaybeASTPassResult(std::vector<ParsedFile> trees);
+
+    bool hasResult() const;
+    std::vector<ParsedFile> &result();
+};
```

Comment:
Another thing you could do here if you wanted:

```
using ParsedFilesOrCancelled = std::optional<std::vector<ParsedFile>>;
```

And then you don't have to re-implement the methods (but also you can't change their names from what they are on `std::optional`. Up to you.